src/Infrastructure/Process/FileCopier.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Infrastructure\Process;
+
+use PhpTuf\ComposerStager\Exception\LogicException;
+use PhpTuf\ComposerStager\Exception\ProcessFailedException;
+use PhpTuf\ComposerStager\Infrastructure\Process\Runner\RsyncRunner;
+
+/**
+ * @internal
+ */
+class FileCopier
```

Comment:
This copier is coupled to `rsync`, so maybe `RsyncFileCopier` or something would be a clearer name for it?

Reply:
Yup. YAGNI. :) I deliberately left it generic until we have another implementation. It'll be trivial to abstract it if/when that time comes.